src/http.c

```diff
@@ -1460,6 +1466,48 @@ http_serve_requests(http_connection_t *hc)
     if ((cmdline = tcp_read_line(hc->hc_fd, &spill)) == NULL)
       goto error;
 
+    // PROXY Protocol v1 support
+    // Format: 'PROXY TCP4 192.168.0.1 192.168.0.11 56324 9981\r\n'
+    //                     SRC-ADDRESS DST-ADDRESS  SPORT DPORT
+    //
+    if ((config.proxy) && (strlen(cmdline) >= 6) && (strncmp(cmdline,"PROXY ",6) == 0 )) {
+        tvhinfo(LS_HTTP, "[PROXY] PROXY protocol detected! cmdline='%s'",cmdline);
+
+        char* pl = cmdline + 6;
+
+        if ((cmdline = tcp_read_line(hc->hc_fd, &spill)) == NULL) {
+            goto error;  // No more data after the PROXY protocol
+        }
+
+        if ( (strlen(pl) >= 7) && (strncmp(pl,"UNKNOWN",7) == 0))
+            goto error;  // Unknown PROXY protocol
+        
+        if ( (strlen(pl) < 5) || (strncmp(pl,"TCP4 ",5) != 0))
+            goto error;  // Only IPv4 supported
+        pl += 5;
+
+        // Check the SRC-ADDRESS
+        c = pl;
+        char ch;
+        for ( ;; ) {
+            if (strlen(pl) == 0) goto error;  // Incomplete PROXY format
```

Comment:
All comments are acceptable.
However, our goal is implementing the code "safe" (=without errors or memory leaks) and "clear" (=easy to read and understand).